Web Inspector: reduce amount of shared code in InspectorNetworkAgent and InspectorPageAgent

burg opened the pull request:
#### 680ba7115028f728b2ffb3c00fb615dfc6805ffa
<pre>
Web Inspector: reduce amount of shared code in InspectorNetworkAgent and InspectorPageAgent
<a href="https://bugs.webkit.org/show_bug.cgi?id=300051">https://bugs.webkit.org/show_bug.cgi?id=300051</a>

Reviewed by Devin Rousso.

In preparation for Site Isolation refactorings in these agents, move static helper
functions, nested classes, and other shared code out of the agent files where possible.
Method implementations have been left alone for now. Many of them need to be improved
in order to pass SaferCPP checks; this work is tracked by <a href="https://webkit.org/b/300125.">https://webkit.org/b/300125.</a>

For site isolation, we will need to coordinate the intercept structs between UIProcess
and web content processes. The same infrastructure will be used to implement network
interception for WebDriver Bidi. So it moves now, despite being used from only NetworkAgent.

* Source/JavaScriptCore/inspector/ContentSearchUtilities.h:
* Source/WebCore/SaferCPPExpectations/UncheckedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedLocalVarsCheckerExpectations:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/inspector/InspectorAuditResourcesObject.cpp:
(WebCore::InspectorAuditResourcesObject::getResources):
(WebCore::InspectorAuditResourcesObject::getResourceContent):
* Source/WebCore/inspector/InspectorNetworkIntercept.cpp: Added.
(Inspector::Intercept::matches):
* Source/WebCore/inspector/InspectorNetworkIntercept.h: Added.
(Inspector::Intercept::operator== const):
(Inspector::PendingInterceptRequest::PendingInterceptRequest):
(Inspector::PendingInterceptRequest::continueWithOriginalRequest):
(Inspector::PendingInterceptRequest::continueWithRequest):
(Inspector::PendingInterceptResponse::PendingInterceptResponse):
(Inspector::PendingInterceptResponse::~PendingInterceptResponse):
(Inspector::PendingInterceptResponse::originalResponse):
(Inspector::PendingInterceptResponse::respondWithOriginalResponse):
(Inspector::PendingInterceptResponse::respond):
* Source/WebCore/inspector/InspectorResourceType.h: Added.
* Source/WebCore/inspector/InspectorResourceUtilities.cpp: Added.
(Inspector::ResourceUtilities::resourceTypeToProtocol):
(Inspector::ResourceUtilities::decodeBuffer):
(Inspector::ResourceUtilities::dataContent):
(Inspector::ResourceUtilities::sharedBufferContent):
(Inspector::ResourceUtilities::mainResourceContent):
(Inspector::ResourceUtilities::resourceContent):
(Inspector::ResourceUtilities::sourceMapURLForResource):
(Inspector::ResourceUtilities::cachedResource):
(Inspector::ResourceUtilities::inspectorResourceType):
(Inspector::ResourceUtilities::cachedResourceTypeToProtocol):
(Inspector::ResourceUtilities::findFrameWithSecurityOrigin):
(Inspector::ResourceUtilities::assertDocumentLoader):
(Inspector::ResourceUtilities::shouldTreatAsText):
(Inspector::ResourceUtilities::createTextDecoder):
(Inspector::ResourceUtilities::textContentForCachedResource):
(Inspector::ResourceUtilities::cachedResourceContent):
* Source/WebCore/inspector/InspectorResourceUtilities.h: Added.
* Source/WebCore/inspector/InspectorStyleSheet.cpp:
(WebCore::InspectorStyleSheet::resourceStyleSheetText const):
* Source/WebCore/inspector/InspectorThreadableLoaderClient.cpp: Added.
(Inspector::InspectorThreadableLoaderClient::didReceiveResponse):
(Inspector::InspectorThreadableLoaderClient::didReceiveData):
(Inspector::InspectorThreadableLoaderClient::didFinishLoading):
(Inspector::InspectorThreadableLoaderClient::didFail):
(Inspector::InspectorThreadableLoaderClient::setLoader):
(Inspector::InspectorThreadableLoaderClient::dispose):
* Source/WebCore/inspector/InspectorThreadableLoaderClient.h: Added.
* Source/WebCore/inspector/NetworkResourcesData.cpp:
(WebCore::NetworkResourcesData::resourceCreated):
(WebCore::NetworkResourcesData::responseReceived):
(WebCore::NetworkResourcesData::setResourceType):
(WebCore::NetworkResourcesData::resourceType):
* Source/WebCore/inspector/NetworkResourcesData.h:
(WebCore::NetworkResourcesData::ResourceData::type const):
(WebCore::NetworkResourcesData::ResourceData::setType):
* Source/WebCore/inspector/agents/InspectorDOMAgent.h:
* Source/WebCore/inspector/agents/InspectorDOMStorageAgent.cpp:
(WebCore::InspectorDOMStorageAgent::findStorageArea):
* Source/WebCore/inspector/agents/InspectorIndexedDBAgent.cpp:
(WebCore::InspectorIndexedDBAgent::requestDatabaseNames):
(WebCore::InspectorIndexedDBAgent::requestDatabase):
(WebCore::InspectorIndexedDBAgent::requestData):
(WebCore::InspectorIndexedDBAgent::clearObjectStore):
* Source/WebCore/inspector/agents/InspectorNetworkAgent.cpp:
(WebCore::InspectorNetworkAgent::buildObjectForCachedResource):
(WebCore::InspectorNetworkAgent::willSendRequest):
(WebCore::resourceTypeForCachedResource):
(WebCore::resourceTypeForLoadType):
(WebCore::InspectorNetworkAgent::didReceiveResponse):
(WebCore::InspectorNetworkAgent::didFinishLoading):
(WebCore::InspectorNetworkAgent::didFailLoading):
(WebCore::InspectorNetworkAgent::didReceiveScriptResponse):
(WebCore::InspectorNetworkAgent::didReceiveThreadableLoaderResponse):
(WebCore::InspectorNetworkAgent::willDestroyCachedResource):
(WebCore::networkStageFromProtocol):
(WebCore::InspectorNetworkAgent::shouldIntercept):
(WebCore::InspectorNetworkAgent::getResponseBody):
(WebCore::InspectorNetworkAgent::addInterception):
(WebCore::InspectorNetworkAgent::removeInterception):
(WebCore::textContentForResourceData):
(WebCore::InspectorNetworkAgent::shouldTreatAsText): Deleted.
(WebCore::InspectorNetworkAgent::createTextDecoder): Deleted.
(WebCore::InspectorNetworkAgent::textContentForCachedResource): Deleted.
(WebCore::InspectorNetworkAgent::cachedResourceContent): Deleted.
(WebCore::InspectorNetworkAgent::Intercept::matches): Deleted.
* Source/WebCore/inspector/agents/InspectorNetworkAgent.h:
(WebCore::InspectorNetworkAgent::PendingInterceptRequest::PendingInterceptRequest): Deleted.
(WebCore::InspectorNetworkAgent::PendingInterceptRequest::continueWithOriginalRequest): Deleted.
(WebCore::InspectorNetworkAgent::PendingInterceptRequest::continueWithRequest): Deleted.
(WebCore::InspectorNetworkAgent::PendingInterceptResponse::PendingInterceptResponse): Deleted.
(WebCore::InspectorNetworkAgent::PendingInterceptResponse::~PendingInterceptResponse): Deleted.
(WebCore::InspectorNetworkAgent::PendingInterceptResponse::originalResponse): Deleted.
(WebCore::InspectorNetworkAgent::PendingInterceptResponse::respondWithOriginalResponse): Deleted.
(WebCore::InspectorNetworkAgent::PendingInterceptResponse::respond): Deleted.
(WebCore::InspectorNetworkAgent::Intercept::operator== const): Deleted.
* Source/WebCore/inspector/agents/InspectorPageAgent.cpp:
(WebCore::allResourcesURLsForFrame):
(WebCore::InspectorPageAgent::getResourceContent):
(WebCore::InspectorPageAgent::searchInResource):
(WebCore::InspectorPageAgent::searchInResources):
(WebCore::InspectorPageAgent::buildObjectForFrameTree):
(WebCore::decodeBuffer): Deleted.
(WebCore::InspectorPageAgent::mainResourceContent): Deleted.
(WebCore::InspectorPageAgent::sharedBufferContent): Deleted.
(WebCore::InspectorPageAgent::dataContent): Deleted.
(): Deleted.
(WebCore::InspectorPageAgent::resourceContent): Deleted.
(WebCore::InspectorPageAgent::sourceMapURLForResource): Deleted.
(WebCore::InspectorPageAgent::cachedResource): Deleted.
(WebCore::InspectorPageAgent::resourceTypeJSON): Deleted.
(WebCore::InspectorPageAgent::inspectorResourceType): Deleted.
(WebCore::InspectorPageAgent::cachedResourceTypeJSON): Deleted.
(WebCore::InspectorPageAgent::findFrameWithSecurityOrigin): Deleted.
(WebCore::InspectorPageAgent::assertDocumentLoader): Deleted.
* Source/WebCore/inspector/agents/InspectorPageAgent.h:
* Source/WebCore/inspector/agents/page/PageDebuggerAgent.cpp:
(WebCore::PageDebuggerAgent::sourceMapURLForScript):

Canonical link: <a href="https://commits.webkit.org/301444@main">https://commits.webkit.org/301444@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f5b3ca125393a3b979f7cdbeff0712eafdbdd797

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/126029 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/45683 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/36459 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/132860 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/77851 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/cf442197-57a7-4db9-90cb-8cbca25257ba) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/127900 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/46352 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/54229 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/132860 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/77851 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/9e13902c-b078-4272-b5b0-c00fca460c4a) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/128977 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/37083 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/112704 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/132860 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/9606c749-2d36-4203-8d42-f410b66a7651) 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/35991 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/135/builds/30890 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🛠 wpe-cairo~~](https://ews-build.webkit.org/#/builders/65/builds/76331 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/118098 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/106864 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/31105 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/135556 "Built successfully") | | 
| [  ~~🛠 🧪 jsc-arm64~~](https://ews-build.webkit.org/#/builders/12/builds/124535 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/52789 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/40521 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/104470 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/53243 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/108918 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/104190 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26534 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/49582 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/27912 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/50168 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/52685 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/58516 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/157546 "Built successfully") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/125/builds/52022 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/39443 "Passed tests") | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/129/builds/55369 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/53724 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

dcrousso commented on Source/WebCore/inspector/InspectorResourceType.h:
```diff
@@ -0,0 +1,48 @@
+
+namespace Inspector {
+
+enum class ResourceType {
```
can/should we give this a `: uint8_t`?

dcrousso commented on Source/WebCore/inspector/InspectorResourceType.h:
```diff
@@ -0,0 +1,48 @@
+namespace Inspector {
+
+enum class ResourceType {
+Document,
```
Style: i think these should all be indented

dcrousso commented on Source/WebCore/inspector/InspectorResourceUtilities.h:
```diff
@@ -0,0 +1,69 @@
+}
+
+namespace Inspector {
+namespace Protocol {
+}
```
is this needed?

dcrousso commented on Source/WebCore/inspector/InspectorNetworkIntercept.h:
```diff
@@ -0,0 +1,127 @@
+
+enum class NetworkStage { Request, Response };
+
+struct Intercept {
```
i dont think `Intercept` or `PendingInterceptRequest` or `PendingInterceptResponse` are (or ever will) be used outside of `InspectorNetworkAgent`.  as such, id recommend we keep them there are they're fundamentally just fancy wrappers around data/callbacks (similar to how the `InspectorDebuggerAgent` has a `BlackboxedScript` or `SymbolicBreakpoint`) and are pretty specific to how the `InspectorNetworkAgent` and `Network` protocol work

burg replied:
for site isolation, we will need to coordinate the intercept structs between UIProcess and webprocesses. And for WebDriver Bidi, we will use the same infrastructure to implement its version of network intercepts.

dcrousso commented on Source/WebCore/inspector/InspectorResourceUtilities.cpp:
```diff
@@ -0,0 +1,369 @@
+
+using namespace WebCore;
+
+Inspector::Protocol::Page::ResourceType resourceTypeToProtocol(Inspector::ResourceType resourceType)
```
just to help me understand, were there any significant changes made to any of these methods or were they just a simple cut-paste-rename from `InspectorNetworkAgent` and `InspectorPageAgent`?

burg replied:
copy-paste, adopt enum class, and rename resourceTypeJSON -> resourceTypeToProtocol.

dcrousso commented on Source/WebCore/inspector/InspectorNetworkIntercept.h:
```diff
@@ -0,0 +1,127 @@
+
+namespace Inspector {
+
+enum class NetworkStage { Request, Response };
```
should we give this a `: bool`?

dcrousso reviewed (approved):
neat!  one request about some of the internal `InspectorNetworkAgent` classes not being moved outside, but otherwise this looks great!

burg committed:
Web Inspector: reduce amount of shared code in InspectorNetworkAgent and InspectorPageAgent
https://bugs.webkit.org/show_bug.cgi?id=300051

Reviewed by Devin Rousso.

In preparation for Site Isolation refactorings in these agents, move static helper
functions, nested classes, and other shared code out of the agent files where possible.
Method implementations have been left alone for now. Many of them need to be improved
in order to pass SaferCPP checks; this work is tracked by https://webkit.org/b/300125.

For site isolation, we will need to coordinate the intercept structs between UIProcess
and web content processes. The same infrastructure will be used to implement network
interception for WebDriver Bidi. So it moves now, despite being used from only NetworkAgent.

* Source/JavaScriptCore/inspector/ContentSearchUtilities.h:
* Source/WebCore/SaferCPPExpectations/UncheckedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedLocalVarsCheckerExpectations:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/inspector/InspectorAuditResourcesObject.cpp:
(WebCore::InspectorAuditResourcesObject::getResources):
(WebCore::InspectorAuditResourcesObject::getResourceContent):
* Source/WebCore/inspector/InspectorNetworkIntercept.cpp: Added.
(Inspector::Intercept::matches):
* Source/WebCore/inspector/InspectorNetworkIntercept.h: Added.
(Inspector::Intercept::operator== const):
(Inspector::PendingInterceptRequest::PendingInterceptRequest):
(Inspector::PendingInterceptRequest::continueWithOriginalRequest):
(Inspector::PendingInterceptRequest::continueWithRequest):
(Inspector::PendingInterceptResponse::PendingInterceptResponse):
(Inspector::PendingInterceptResponse::~PendingInterceptResponse):
(Inspector::PendingInterceptResponse::originalResponse):
(Inspector::PendingInterceptResponse::respondWithOriginalResponse):
(Inspector::PendingInterceptResponse::respond):
* Source/WebCore/inspector/InspectorResourceType.h: Added.
* Source/WebCore/inspector/InspectorResourceUtilities.cpp: Added.
(Inspector::ResourceUtilities::resourceTypeToProtocol):
(Inspector::ResourceUtilities::decodeBuffer):
(Inspector::ResourceUtilities::dataContent):
(Inspector::ResourceUtilities::sharedBufferContent):
(Inspector::ResourceUtilities::mainResourceContent):
(Inspector::ResourceUtilities::resourceContent):
(Inspector::ResourceUtilities::sourceMapURLForResource):
(Inspector::ResourceUtilities::cachedResource):
(Inspector::ResourceUtilities::inspectorResourceType):
(Inspector::ResourceUtilities::cachedResourceTypeToProtocol):
(Inspector::ResourceUtilities::findFrameWithSecurityOrigin):
(Inspector::ResourceUtilities::assertDocumentLoader):
(Inspector::ResourceUtilities::shouldTreatAsText):
(Inspector::ResourceUtilities::createTextDecoder):
(Inspector::ResourceUtilities::textContentForCachedResource):
(Inspector::ResourceUtilities::cachedResourceContent):
* Source/WebCore/inspector/InspectorResourceUtilities.h: Added.
* Source/WebCore/inspector/InspectorStyleSheet.cpp:
(WebCore::InspectorStyleSheet::resourceStyleSheetText const):
* Source/WebCore/inspector/InspectorThreadableLoaderClient.cpp: Added.
(Inspector::InspectorThreadableLoaderClient::didReceiveResponse):
(Inspector::InspectorThreadableLoaderClient::didReceiveData):
(Inspector::InspectorThreadableLoaderClient::didFinishLoading):
(Inspector::InspectorThreadableLoaderClient::didFail):
(Inspector::InspectorThreadableLoaderClient::setLoader):
(Inspector::InspectorThreadableLoaderClient::dispose):
* Source/WebCore/inspector/InspectorThreadableLoaderClient.h: Added.
* Source/WebCore/inspector/NetworkResourcesData.cpp:
(WebCore::NetworkResourcesData::resourceCreated):
(WebCore::NetworkResourcesData::responseReceived):
(WebCore::NetworkResourcesData::setResourceType):
(WebCore::NetworkResourcesData::resourceType):
* Source/WebCore/inspector/NetworkResourcesData.h:
(WebCore::NetworkResourcesData::ResourceData::type const):
(WebCore::NetworkResourcesData::ResourceData::setType):
* Source/WebCore/inspector/agents/InspectorDOMAgent.h:
* Source/WebCore/inspector/agents/InspectorDOMStorageAgent.cpp:
(WebCore::InspectorDOMStorageAgent::findStorageArea):
* Source/WebCore/inspector/agents/InspectorIndexedDBAgent.cpp:
(WebCore::InspectorIndexedDBAgent::requestDatabaseNames):
(WebCore::InspectorIndexedDBAgent::requestDatabase):
(WebCore::InspectorIndexedDBAgent::requestData):
(WebCore::InspectorIndexedDBAgent::clearObjectStore):
* Source/WebCore/inspector/agents/InspectorNetworkAgent.cpp:
(WebCore::InspectorNetworkAgent::buildObjectForCachedResource):
(WebCore::InspectorNetworkAgent::willSendRequest):
(WebCore::resourceTypeForCachedResource):
(WebCore::resourceTypeForLoadType):
(WebCore::InspectorNetworkAgent::didReceiveResponse):
(WebCore::InspectorNetworkAgent::didFinishLoading):
(WebCore::InspectorNetworkAgent::didFailLoading):
(WebCore::InspectorNetworkAgent::didReceiveScriptResponse):
(WebCore::InspectorNetworkAgent::didReceiveThreadableLoaderResponse):
(WebCore::InspectorNetworkAgent::willDestroyCachedResource):
(WebCore::networkStageFromProtocol):
(WebCore::InspectorNetworkAgent::shouldIntercept):
(WebCore::InspectorNetworkAgent::getResponseBody):
(WebCore::InspectorNetworkAgent::addInterception):
(WebCore::InspectorNetworkAgent::removeInterception):
(WebCore::textContentForResourceData):
(WebCore::InspectorNetworkAgent::shouldTreatAsText): Deleted.
(WebCore::InspectorNetworkAgent::createTextDecoder): Deleted.
(WebCore::InspectorNetworkAgent::textContentForCachedResource): Deleted.
(WebCore::InspectorNetworkAgent::cachedResourceContent): Deleted.
(WebCore::InspectorNetworkAgent::Intercept::matches): Deleted.
* Source/WebCore/inspector/agents/InspectorNetworkAgent.h:
(WebCore::InspectorNetworkAgent::PendingInterceptRequest::PendingInterceptRequest): Deleted.
(WebCore::InspectorNetworkAgent::PendingInterceptRequest::continueWithOriginalRequest): Deleted.
(WebCore::InspectorNetworkAgent::PendingInterceptRequest::continueWithRequest): Deleted.
(WebCore::InspectorNetworkAgent::PendingInterceptResponse::PendingInterceptResponse): Deleted.
(WebCore::InspectorNetworkAgent::PendingInterceptResponse::~PendingInterceptResponse): Deleted.
(WebCore::InspectorNetworkAgent::PendingInterceptResponse::originalResponse): Deleted.
(WebCore::InspectorNetworkAgent::PendingInterceptResponse::respondWithOriginalResponse): Deleted.
(WebCore::InspectorNetworkAgent::PendingInterceptResponse::respond): Deleted.
(WebCore::InspectorNetworkAgent::Intercept::operator== const): Deleted.
* Source/WebCore/inspector/agents/InspectorPageAgent.cpp:
(WebCore::allResourcesURLsForFrame):
(WebCore::InspectorPageAgent::getResourceContent):
(WebCore::InspectorPageAgent::searchInResource):
(WebCore::InspectorPageAgent::searchInResources):
(WebCore::InspectorPageAgent::buildObjectForFrameTree):
(WebCore::decodeBuffer): Deleted.
(WebCore::InspectorPageAgent::mainResourceContent): Deleted.
(WebCore::InspectorPageAgent::sharedBufferContent): Deleted.
(WebCore::InspectorPageAgent::dataContent): Deleted.
(): Deleted.
(WebCore::InspectorPageAgent::resourceContent): Deleted.
(WebCore::InspectorPageAgent::sourceMapURLForResource): Deleted.
(WebCore::InspectorPageAgent::cachedResource): Deleted.
(WebCore::InspectorPageAgent::resourceTypeJSON): Deleted.
(WebCore::InspectorPageAgent::inspectorResourceType): Deleted.
(WebCore::InspectorPageAgent::cachedResourceTypeJSON): Deleted.
(WebCore::InspectorPageAgent::findFrameWithSecurityOrigin): Deleted.
(WebCore::InspectorPageAgent::assertDocumentLoader): Deleted.
* Source/WebCore/inspector/agents/InspectorPageAgent.h:
* Source/WebCore/inspector/agents/page/PageDebuggerAgent.cpp:
(WebCore::PageDebuggerAgent::sourceMapURLForScript):

Canonical link: https://commits.webkit.org/301444@main
